src/PointwiseFunctions/AnalyticSolutions/WaveEquation/RegularSphericalWave.hpp

```diff
@@ -0,0 +1,104 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines ScalarWave::Solutions::RegularSphericalWave
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/MathFunctions/MathFunction.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace ScalarWave {
+struct Pi;
+struct Psi;
+template <size_t Dim>
+struct Phi;
+}  // namespace ScalarWave
+namespace Tags {
+template <typename Tag>
+struct dt;
+}  // namespace Tags
+template <size_t VolumeDim>
+class MathFunction;
+
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace ScalarWave {
+namespace Solutions {
+/*!
+ * \brief A spherical wave solution to the Euclidean wave equation that is
+ * regular at the origin
+ *
+ * The solution is given by \f$\Psi(\vec{x},t) = \Psi(r,t) =
+ * \frac{F(r-t)-F(r+t)}{r}\f$ describing an outgoing and an ingoing wave
+ * with profile \f$F(u)\f$ that meet (and cancel each other) at \f$t=0\f$. The
+ * solution is approximated by its Taylor expansion
+ * \f$\Psi(r,t)=\left[F^\prime(r-t)-F^\prime(r+t)\right]_{r=0} +
+ * \frac{r}{2}\left[F^{\prime\prime}(r-t)-F^{\prime\prime}(r+t)\right]_{r=0} +
+ * \mathcal{O}(r^2)\f$ for small $r$. This expansion is currently employed only
+ * at the point \f$r=0\f$ (determined by `equal_within_roundoff`) and therefore
+ * only evaluated to zeroth order.
+ *
+ * \requires the profile \f$F(u)\f$ to have a length scale of order unity so
+ * that _small \f$r\f$_ means \f$r\ll 1\f$. This is without loss of generality
+ * because of the scale invariance of the wave equation. The profile could be a
+ * Gausssian with width 1, for instance.
+ *
+ * \tparam Dim The spatial dimension of the solution. Currently only implemented
+ * for 3 spatial dimensions.
+ */
+template <size_t Dim>
```

Comment:
why is this a template?

Reply:
This is not consistent with other code.  See EinsteinSolutions/KerrSchild which is not templated on dimension.  It is quite clear it only works in 3D as it expects 3-d spatial coordinates.

Reply:
I'm with @kidder on this.

Reply:
Ok, fixed